### PR TITLE
Add water year capability

### LINKED ISSUE #8931: Add ability to define water seasons in weather sampler
### Describe the new feature

The weather sampler is limited to changing sample years only on day 1 - right in the middle of a summer rainfall season. It would be useful to allow a season break to be specified (eg 1-oct) so that an entire water season is used, not half of one, half the other.

### PR DESCRIPTION
**Pull request by peter-devoil**
resolves #8931

### CODE REVIEW

**Comment by peter-devoil** [comment]
retest this please jenkins

**Comment by peter-devoil** [comment]
This is ready, @hol353

File: Models/Climate/WeatherSampler.cs [hunk 1]
```diff
@@ -86,6 +86,10 @@ public enum RandomiserTypeEnum
         /// <summary>Is 'specify years' enabled?</summary>
         public bool IsSpecifyYearsEnabled { get { return TypeOfSampling == RandomiserTypeEnum.SpecificYears; } }
 
+        /// <summary>The date when years tick over.</summary>
+        [Summary]
+        [Description("Water year day 0 - when to split the year from one to the next (d-mmm)")]
+        public string WaterYear { get; set; }
```
Comment (hol353) [inline Models/Climate/WeatherSampler.cs:92]: I'm not sure what water years are. Can you just have the description something like: ```The date to being sampling years (d-mmm). Leave blank for 1-Jan```

Perhaps call the property SplitDate?

**Comment by hol353** [comment]
@peter-devoil Have you had a look at my comment above?

**Pull request merged**